apps/arweave/src/ar_mining_worker.erl

```diff
@@ -752,11 +752,9 @@ update_sub_chunk_cache_size(0, _SessionKey, State) ->
 	State;
 update_sub_chunk_cache_size(Delta, SessionKey, State) ->
 	CacheSize = maps:get(SessionKey, State#state.sub_chunk_cache_size, 0),
-	prometheus_gauge:inc(mining_server_chunk_cache_size,
-			[State#state.partition_number], Delta),
-	State#state{
-		sub_chunk_cache_size = maps:put(SessionKey, CacheSize + Delta,
-		State#state.sub_chunk_cache_size) }.
+	NewCacheSize = ar_util:clamp_lower(CacheSize + Delta, 0),
```

Comment:
The key change: do not let the sub_chunk_cache_size go below zero, as this does not make any sense.

Reply:
Aren't we fighting the symptom? Perhaps, let's leave a comment this is a temporary measure until we figure out why we subtract more than we add, if that is the case.

Reply:
@shizzard what about adding a log_warning message whenever CacheSize+Delta is below zero? We do want to fix the OOMs as quick as possible and clamping may help there, but I'm a little worried we may also lose this valuable signal that something is wrong. Which may make tracking down the true root cause harder.

Reply:
I slacked with some more information that might help with reproducing the negative cache size (and so might help get at the root cause)

Reply:
> what about adding a log_warning message

Thats a very good idea, will do.